modules/nextflow/src/main/groovy/nextflow/ast/NextflowDSLImpl.groovy

```diff
@@ -263,7 +257,7 @@ class NextflowDSLImpl implements ASTTransformation {
                 stm.setExpression(loadCall)
             }
             if( isInvalidVariableAssignment(stm) ) {
-                syntaxError(stm, 'Cannot declare a script variable with the same name as the script')
+                syntaxError(stm, 'Cannot declare a variable identifier with the same name as the script file')
```

Comment:
"Script variable" would be more precise because local variables are still allowed. Note from the tests that you need only add a `def` to make it work.

Actually we should probably add that suggestion to the error message...

Reply:
yeah, that's true. just because don't like the repetition "script variable" and "script file"